### DISCUSSION TITLE
Copy constructor and uniform initialization

### DISCUSSION THREAD

**Discussion by me21** [nlohmann/json#3347] [Q&A] [answered]
Hello,

I have a question which I'm currently having a hard time to understand. I compile my project in C++17 mode with mingw-gcc 11.2.0.
The following code
```
json j1{{"one", "two}};
json j2{j1};
std::cout << j2.dump() << std::endl;
```
gives `[{"one":"two"}]` (notice extra square brackets), whereas
```
json j1{{"one", "two}};
json j2(j1);
std::cout << j2.dump() << std::endl;
```
gives `{"one":"two"}`.

I thought that the first snippet should invoke copy constructor too, but apparently I'm wrong. Please help me understand what's going on there with respect to C++ standard.

**Comment by nlohmann** [comment, marked as answer]
See https://json.nlohmann.me/home/faq/#brace-initialization-yields-arrays

**Reply by me21** [reply]
Yeah, found the corresponding part in the C++ standard: https://en.cppreference.com/w/cpp/language/list_initialization

Basically, if the class has a constructor which takes initializer_list, it will be called instead of a copy constructor (if using curly braces).
